packages/common/src/project/utils.ts

```diff
@@ -48,16 +48,24 @@ export function validateSemver(current: string, required: string): boolean {
   return satisfies(current, required, {includePrerelease: true});
 }
 
+@ValidatorConstraint({name: 'semver', async: false})
 export class SemverVersionValidator implements ValidatorConstraintInterface {
   validate(value: string | null | undefined): boolean {
-    const validated = valid(value, {includePrerelease: false});
-    if (validated === null || validated === undefined) {
+    if (valid(value, {includePrerelease: false}) === null) {
+      if (validRange(value, {includePrerelease: false}) === null) {
+        return false;
+      } else {
+        return true;
+      }
```

Comment:
It seems strange to me that `validRange` would return `null`, shouldn't it be a boolean?
```suggestion
      return validRange(value, {includePrerelease: false}) !== null;
```

The same change can be applied with the `prerelease` check